pkg/kubectl/scheme/sparse_test.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheme
+
+import (
+	"testing"
+
+	"k8s.io/api/batch/v1"
+	"k8s.io/api/batch/v1beta1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/diff"
+)
+
+func TestCronJob(t *testing.T) {
+	src := &v1beta1.CronJob{ObjectMeta: metav1.ObjectMeta{Name: "foo"}}
+
+	encoder := Codecs.LegacyCodec(v1.SchemeGroupVersion, v1beta1.SchemeGroupVersion)
+	cronjobBytes, err := runtime.Encode(encoder, src)
```

Comment:
the issues with sparse encoding typically came when encoding an internal version, e.g. &batch.CronJob{}

this test is fine, but doesn't really exercise the failing case. since we're excising the internal->"find an external version to encode" paths in kubectl, this test is sufficient to cover the cases we care about